File: planik-android/app/src/main/java/pl/planik/domain/source/UserLocalSource.kt
package pl.planik.domain.source

import pl.planik.domain.model.NewUser
import pl.planik.domain.model.User

interface UserLocalSource {
  fun currentUserId(): Int?
  fun hasUser(): Boolean
  fun hasPlan(): Boolean
  fun setHasPlan(value: Boolean)
  suspend fun getCurrentUser(): User?
  suspend fun createUser(newUser: NewUser): User?
}

File: planik-flutter/planik/lib/domain/source/user_local_source.dart
import 'package:fpdart/fpdart.dart';
import 'package:planik/domain/failure/general_failure.dart';
import 'package:planik/domain/model/user.dart';

abstract class UserLocalSource {
  TaskEither<GeneralFailure, int> currentUserId();

  TaskEither<GeneralFailure, bool> hasUser();

  TaskEither<GeneralFailure, bool> hasPlan();

  TaskEither<GeneralFailure, Unit> setHasPlan({required bool hasPlan});

  TaskEither<GeneralFailure, User> create({required String nickname, required String uid});

  TaskEither<GeneralFailure, User> getCurrentUser();
}
